content/docs/guides/health_checks/health_probes.md

```diff
@@ -13,17 +13,18 @@ Implementing [health probes](https://kubernetes.io/docs/tasks/configure-pod-cont
 
 Because OSM reconfigures application Pods to redirect all incoming and outgoing network traffic through the proxy sidecar, `httpGet` and `tcpSocket` health probes invoked by the kubelet will fail due to the lack of any mTLS context required by the proxy.
 
-For health probes to continue to work as expected from within the mesh, OSM adds configuration to expose the probe endpoint via the proxy and rewrites the probe definitions for new Pods to refer to the proxy-exposed endpoint. All of the functionality of the original probe is still used, OSM simply fronts it with the proxy so the kubelet can communicate with it.
+For `httpGet` health probes to continue to work as expected from within the mesh, OSM adds configuration to expose the probe endpoint via the proxy and rewrites the probe definitions for new Pods to refer to the proxy-exposed endpoint. All of the functionality of the original probe is still used, OSM simply fronts it with the proxy so the kubelet can communicate with it.
 
-For HTTP probes, the following table shows the modified path and port for each probe type.
+To support `tcpSocket` health probes in the mesh special configuration is required. Since OSM redirects all network traffic through Envoy, all ports appear open. Therefore, all TCP connections routed to Pod's injected with an Envoy sidecar will succeed. For `tcpSocket` health probes to fail and succeed as expected in the mesh, OSM rewrites the probes to be `httpGet` probes and adds an IP table command to bypass the Envoy proxy at the `osm-healthcheck` exposed endpoint. The `osm-healthcheck` container is added to the Pod and handles the HTTP health probe requests from kubelet. The handler gets the original TCP port from the request's `Original-Tcp-port` header and attempts to open a socket on the specified port. The response status code for the `httpGet` probe is determined by the success of the TCP connection.
```

Comment:
```suggestion
A special configuration is required to support `tcpSocket` health probes in the mesh. This is because OSM redirects all network traffic through Envoy, so all ports will appear open. This causes all TCP connections routed to Pod's injected with an Envoy sidecar to appear as successful. For `tcpSocket` health probes to work as expected in the mesh, OSM rewrites the probes to be `httpGet` probes and adds an IP table command to bypass the Envoy proxy at the `osm-healthcheck` exposed endpoint. The `osm-healthcheck` container is added to the Pod and handles the HTTP health probe requests from kubelet. The handler gets the original TCP port from the request's `Original-Tcp-port` header and attempts to open a socket on the specified port. The response status code for the `httpGet` probe will reflect if the TCP connection was successfull. ```